app/Http/Controllers/API/v1/ImportController.php

```diff
@@ -0,0 +1,303 @@
+            ),
+            new OA\Response(
+                response: 401,
+                description: 'Unauthorized'
+            ),
+            new OA\Response(
+                response: 500,
+                description: 'Server error'
+            ),
+        ]
+    )]
+    public function getImports(Request $request): JsonResponse
+    {
+        $user = $request->user();
+        $imports = $user->fileImports()->orderBy('id', 'desc')->get();
+
+        return $this->success($imports);
+    }
+
+    #[OA\Put(
+        path: '/imports/{id}/fix',
+        summary: 'Fix records that failed to import',
+        requestBody: new OA\RequestBody(
+            required: true,
+            content: new OA\JsonContent(
+                required: ['id', 'amount', 'type', 'date'],
+                type: 'array',
+                items: new OA\Items(
+                    '#/components/schemas/FailedImport'
+                )
+            )
+        ),
+        tags: ['Import'],
+        parameters: [
+            new OA\Parameter(
+                name: 'id',
+                description: 'Id of the import instance',
+                in: 'path',
+                required: true,
+                schema: new OA\Schema(type: 'integer')
+            ),
+        ],
+        responses: [
+            new OA\Response(
+                response: 200,
+                description: 'Successful response',
+            ),
+            new OA\Response(
+                response: 206,
+                description: 'Some imports could not be fixed',
+                content: new OA\JsonContent(type: 'array',
+                    items: new OA\Items('#/components/schemas/FailedImport')
+                )
+            ),
+            new OA\Response(
+                response: 401,
+                description: 'Unauthorized'
+            ),
+            new OA\Response(
+                response: 500,
+                description: 'Server error'
+            ),
+        ]
+    )]
+    public function fixFailedImports(FixFailedImportsRequest $request, string $id): JsonResponse
+    {
+        $user = $request->user();
+        $import = $user->fileImports()->find($id);
+        if (is_null($import)) {
+            return $this->failure('File import instance not found', 404);
+        }
+        $failedImportsToReturn = [];
+        $data = $request->all();
+        for ($i = 0; $i < count($data); $i++) {
+            // verify if this failed import belong to this request
+            $importToFix = $data[$i];
+            $failedImport = $import->failedImports()->find($importToFix['id']);
+            if (is_null($failedImport)) {
+                $importToFix['reason'] = 'This failed record does not belong to this import instance';
+                $failedImportsToReturn[] = $importToFix;
+            } elseif (! $this->fileImportService->isValidDate($importToFix['date'])) {
+                $importToFix['reason'] = 'Date must be in the format YYYY-MM-DD';
```

Comment:
```suggestion
                $importToFix['reason'] = 'Date must be in the format YYYY-MM-DD';
```

Why are we using this counter-intuitive format? Is not DD-MM-YYYY better?

Reply:
Laravel forces us to use YYYY-MM-DD

Reply:
Maybe at the database level? and not the API level? 

this is not so important though so feel free to mark it as resolved.